app/components/edit/EditAddress.jsx

```diff
@@ -1,7 +1,119 @@
-import React from 'react';
+import React, { useState } from 'react';
+import ReactModal from 'react-modal';
 import PropTypes from 'prop-types';
 import editCollectionHOC from './EditCollection';
 
+import s from './EditAddress.module.scss';
+
+// Subcomponents
+
+/** Modal containing the edit and add new address form.
+ *
+ * The form data is only synchronized with the outer component's state when the
+ * form is submitted. This allows the user to back out of any changes with the
+ * "Cancel" button. Therefore, we have chosen to implement the form inputs as
+ * uncontrolled components.
+ */
+const EditAddressModal = ({
+  isOpen, onRequestClose, defaultData, onSave,
+}) => {
+  const handleSubmit = e => {
+    e.preventDefault();
+    const formData = new FormData(e.currentTarget);
+    onSave(Object.fromEntries(formData.entries()));
+    onRequestClose();
+  };
+  return (
+    <ReactModal
+      className={s.modal}
+      overlayClassName={s.overlay}
+      isOpen={isOpen}
+      onRequestClose={onRequestClose}
+    >
+      <form className={s.modalContent} onSubmit={handleSubmit}>
+        <h1 className={s.title}>Add New Address</h1>
+        <div className={s.formBody}>
+          <label className={s.inputLabel}>
+            Address name (optional)
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="name"
+              placeholder="Name"
+              defaultValue={defaultData.name}
+            />
+          </label>
+          <div className={s.formControlGroup}>
+            <span className={s.formControlGroupLabel}>Street Address</span>
+            <input
```

Comment:
Note to self: Make all of these field required, since these are uncontrolled components.